src/Dialect/Krnl/Krnl.td

```diff
@@ -1036,3 +1036,23 @@ def KrnlFindIndexOp : Op<Krnl_Dialect, "find_index",
   let arguments = (ins AnyTypeOf<[StringType, I64]>:$input, I32MemRef:$G, I32MemRef:$V, I32:$len);
   let results = (outs Index:$index);
 }
+
+def KrnlPrintTensorOp : Op<Krnl_Dialect, "print_tensor", [MemRefsNormalizable]> {
+  let summary = "Print a tensor.";
+  let description = [{
+    This operation can be used to generate a call to a runtime function which prints a tensor.
+  }];
+
+  let arguments = (ins AnyMemRef:$input);
+} 
+
+def KrnlPrintOp : Op<Krnl_Dialect, "print", [MemRefsNormalizable]> {
+  let summary = "Print a value.";
+  let description = [{
+    This operation can be used to print the input value. The user needs to provide a 
+    format string (à la printf) to specify how to print the input value. 
+    If the input value is not specified the operator will print the format string.
+  }];
+
+  let arguments = (ins StrAttr:$format, Optional<AnyType>:$input);
```

Comment:
I feel that input can only be primitive types, not Memref, not struct or list. Is it better to pass the type and let the runtime to generate the format string if the format string is platform dependent (such 32 vs. 64).

Reply:
We lower the KrnlPrintOp to a printf call rather than a call to an API in the onnx-mlir runtime library.  If you want I can create the format string during lowering of the operator.  Some ppl might want to print a long long as a pointer though (so with %p rather than %lld) and the current operator would allow for that scenario.